### ISSUE TITLE
Improve diagnostic for '(args: number) => {}' in JavaScript mode

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#976]
```typescript
let f = (args: number) => {};
```
Expected:
> E0224: TypeScript type annotations are not allowed in JavaScript code

Actual:
> hello.js:1:9: error: unmatched parenthesis [E0056]
> hello.js:1:14: error: missing semicolon after statement [E0027]
> hello.js:1:14: error: unexpected token [E0054]
> hello.js:1:22: error: unmatched parenthesis [E0056]
> hello.js:1:23: error: missing semicolon after statement [E0027]
> hello.js:1:24: error: missing parameters for arrow function [E0105]
> hello.js:1:10: warning: use of undeclared variable: args [E0057]
> hello.js:1:16: warning: use of undeclared variable: number [E0057]

**Comment by leszek888** [comment]
I'd like to tackle this one.
